website/docs/language/import/index.mdx

```diff
@@ -0,0 +1,136 @@
+
+Once imported, an existing resource is present in Terraform's configuration and state, much like any other resource Terraform created. You can then manage that resource like any other, updating its attributes and destroying it as part of a standard resource lifecycle.
+
+The `import` block records that Terraform imported a resource and that Terraform did not previously create that resource with its configuration. Put another way, the `import` block records the user's intent to add an import to Terraform's plan graph.
+
+## Syntax
+
+The `import` block can appear in any Terraform configuration file. A common pattern is to create an `imports.tf` file, or to place each `import` block beside the `resource` block it imports into.
+
+```hcl
+import {
+  to = aws_instance.example
+  id = "i-abcd1234"
+}
+
+resource "aws_instance" "example" {
+  name = "hashi"
+  # (other resource arguments...)
+}
+```
+
+The above `import` block defines an import of the AWS instance with the ID "i-abcd1234" into the `aws_instance.example` resource in the root module.
+
+The `import` block has the following arguments:
+ - `to` - The resource instance address this import is targeting.
+ - `id` - A string with the import ID of the resource.
+ - `provider` (optional) - An optional custom resource provider, see [The Resource provider Meta-Argument](/terraform/language/meta-arguments/resource-provider) for details.
+
+If you do not give the `import` block a `provider` argument, Terraform attempts to import from the default provider by default. 
+
+### Import ID
+
+To locate the resource you want to import, Terraform requires an import ID in the `id` argument. This argument must currently be supplied as a literal string.
+
+The import block's `id` argument is the resource's import ID. The identifier you use for the `id` argument is resource-specific. You can find the required ID in the [provider documentation](https://registry.terraform.io/browse/providers) for the resource you wish to import.
+
+## Plan and Apply an Import
+
+Terraform processes the `import` block during the plan stage. Along with create, update, remove, and move, import is an action Terraform performs on a resource. Once a plan is approved, Terraform imports the resource into its state during the subsequent apply stage.
+
+You can use `import` blocks by doing the following:
+1. Define an `import` block for the resource(s).
+1. Add a `resource` block in your configuration.
+1. Run `terraform plan` to see how Terraform plans to import the resource(s).
+1. Apply the configuration and have Terraform import resources and update your Terraform state.
+
+> **Hands-on:** Try the [State Import](/terraform/tutorials/state/state-import?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial.
+
+The `import` block is [_idempotent_](https://en.wikipedia.org/wiki/Idempotence), meaning that applying an import action and running another plan will not generate another import action. 
```

Comment:
not sure about our conventions for linking to other sources. @rkoron007? 

```suggestion
The `import` block is [_idempotent_](https://en.wikipedia.org/wiki/Idempotence), so the import operation will only run once for any import block.
```

Reply:
I was the one who suggested this link. I think folks not familiar with this word may want further examples of how to use it. It's a perfect word to use here though. 

We don't say anything about linking to outside resources in any of our style guides; and I think Wikipedia is reliable enough to link to here.

Reply:
It's true that the import operation will only run once for any import block--as long as the resource remains in state. I'd be wary of making any more general statement than that. I think we want to say something that means "_ceteris paribus_ applying a config containing an import block will only apply the import once".